checker.c

```diff
@@ -1,21 +1,80 @@
 #include <stdio.h>
 #include <assert.h>
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
+const float min_temp = 0;
+const float max_temp = 45;
+const float min_soc = 20;
+const float max_soc = 80;
+const float max_CR = 0.8;
+
+int tempIsok (float temperature)
+{
+  if(temperature < min_temp || temperature > max_temp)
+  {
     printf("Temperature out of range!\n");
     return 0;
-  } else if(soc < 20 || soc > 80) {
+  }
+  else
+  {
+    printf("Temperature is ok!\n");  
+    return 1;
+  }
+}
+
+int socIsok (float soc)
+{
+  if(soc < min_soc || soc > max_soc)
```

Comment:
These comparisons look the same, with similar range-checks.
This means any change in the comparison would need to be done in three places.
Make a common function to avoid the above.